### PR TITLE
Inwebo Authenticator added

### PR DESCRIPTION
**Pull request by ashalya**

### CODE REVIEW

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            } catch (UnsupportedEncodingException e) {
+                log.error("Error while encoding the input");
+            }
+            JSONParser parser = new JSONParser();
+            try {
+                json = (JSONObject) parser.parse(br);
+            } catch (IOException e) {
+                log.error("Error while parsing the json object ");
+            } catch (ParseException e) {
+                e.printStackTrace();
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Remove the printstackTrace

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                    br = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+                }
+            } catch (UnsupportedEncodingException e) {
+                log.error("Error while encoding the input");
+            }
+            JSONParser parser = new JSONParser();
+            try {
+                json = (JSONObject) parser.parse(br);
+            } catch (IOException e) {
+                log.error("Error while parsing the json object ");
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Log the error with exception

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        String urlParameters = null;
+        JSONObject json = null;
+        try {
+            urlParameters = "action=checkPushResult"
+                    + "&serviceId=" + URLEncoder.encode("" + serviceId, "UTF-8")
+                    + "&userId=" + URLEncoder.encode(login, "UTF-8")
+                    + "&sessionId=" + URLEncoder.encode(sessionId, "UTF-8")
+                    + "&format=json";
+        } catch (UnsupportedEncodingException e1) {
+            e1.printStackTrace();
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Remove printStackTrace

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+
+public class CheckPushResult {
+    private static final Log log = LogFactory.getLog(CheckPushResult.class);
+
+    private static String urlString = "https://api.myinwebo.com/FS?";
+    private String serviceId;
+    private String p12file;
+    private String p12password;
+    SSLContext context = null;
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Add access modifier for this

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                }
+            } catch (IOException e) {
+                log.error("Error while creating the connection" + e.getMessage(), e);
+            }
+            if (conn != null) {
+                conn.setSSLSocketFactory(sslsocketfactory);
+            }
+
+            try {
+                if (conn != null) {
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: if (conn != null) {

this if condition is duplicated in several places. Try to minimize the duplicate codes

Comment (ashalya) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: https://github.com/wso2/is-connectors/pull/4

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        </pluginManagement>
+
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>2.0</version>
+                <configuration>
+                    <source>1.5</source>
+                    <target>1.5</target>
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml]: source version should be 1.7

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Private-Package>org.wso2.carbon.identity.authenticator.internal</Private-Package>
+                        <Import-Package>org.apache.axis2.*,
+                            org.apache.axiom.*,
+                            org.wso2.carbon.ui.*,
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml]: define versions for importing packages

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+                            org.apache.commons.logging.*; version="1.0.4",
+                            org.osgi.framework,
+                            org.wso2.carbon.identity.application.authentication.framework.*,
+                            javax.servlet,
+                            javax.servlet.http,
+                            *;resolution:=optional
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml]: remove "resolution:=optional"

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: avoid using * when import. Import only necessary classes

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml (hunk 6 above)
Comment (ashalya) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/pom.xml]: This connector used the default IS Arch-type. will update soon

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+
+public class CheckPushResult {
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: This is not suitable for a class name.. This sounds more like a method..

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+                    + "&format=json";
+        } catch (UnsupportedEncodingException e1) {
+            log.error("Error while encoding the Url" + e1.getMessage(), e1);
+            json.put(InweboConstants.ERROR, InweboConstants.ERRORCODE);
+        }
+        try {
+            if (this.context == null) {
+                try {
+                    this.context = PushAuthenticate.setHttpsClientCert(this.p12file, this.p12password);
+                } catch (Exception e) {
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Don't catch Exception here. Try to catch most fine grained exception

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+                    log.error("Error while adding the certificate" + e.getMessage(), e);
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: throw exeption here.

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+        this.p12password = p12password;
+    }
+
+    /**
+     * Prompt for a login and an OTP and check if they are OK.
+     */
+    public JSONObject checkPushResult(String userId, String sessionId) {
+        String urlParameters = null;
+        JSONObject json = null;
+        try {
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: don't use several try catch statements here. Try to use try catch statements in more readable manner

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+            urlParameters = "action=checkPushResult"
+                    + "&serviceId=" + URLEncoder.encode("" + serviceId, InweboConstants.ENCODING)
+                    + "&userId=" + URLEncoder.encode(userId, InweboConstants.ENCODING)
+                    + "&sessionId=" + URLEncoder.encode(sessionId, InweboConstants.ENCODING)
+                    + "&format=json";
+        } catch (UnsupportedEncodingException e1) {
+            log.error("Error while encoding the Url" + e1.getMessage(), e1);
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Is it meaningful to proceed in the flow if this encoding error happens?

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+                is = conn.getInputStream();
+                BufferedReader br = null;
+                if (is != null) {
+                    br = new BufferedReader(new InputStreamReader(is, InweboConstants.ENCODING));
+                }
+                JSONParser parser = new JSONParser();
+                json = (JSONObject) parser.parse(br);
+
+            } catch (MalformedURLException e) {
+                log.error("Error while creating the URL" + e.getMessage(), e);
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: throw exception here

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+            } catch (IOException e) {
+                log.error("Error while creating the connection" + e.getMessage(), e);
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: throw exception here

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+            } catch (ParseException e) {
+                log.error("Error while parsing the json object " + e.getMessage(), e);
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: throw exception here

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushREST.java [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+        int retry = 0;
+        while ((retry < retryCount)) {
+            retry++;
+            result = cr.checkPushResult(userId, SessionId);
+            if (!result.get(InweboConstants.ERROR).equals(InweboConstants.CODEWAITTING)) break;
+            try {
+                log.info("request pending...  " + result);
+                Thread.sleep(retryInterval);
+            } catch (InterruptedException e) {
+                log.error("Error while getting response" + e.getMessage(), e);
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushREST.java]: throw exception here

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+            }
+        } catch (Exception e) {
+            log.error("Error while checking push result");
+            json.put(InweboConstants.ERROR, InweboConstants.ERRORCODE);
+        }
+        return json;
+    }
+}
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java]: Final block needs to be there to close streams..

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushREST.java [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+        this.serviceId = serviceId;
+        this.p12file = p12file;
+        this.p12password = p12password;
+        this.userId = userId;
+        this.retryCount = retryCount;
+        this.retryInterval = retryInterval;
+    }
+
+    public String pushRESTCall() {
+        String SessionId;
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushREST.java]: use first letter simple in variables like sessionId

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 21]
```diff
@@ -0,0 +1,240 @@
...
+                    if (userId != null && serviceId != null && p12file != null && p12password != null) {
+                        PushREST push;
+                        push = new PushREST(serviceId, p12file, p12password, userId, retryCount, retryInterval);
+                        pushResponse = push.run();
+                        PushREST.showServlet(response, relyingParty, type, finalReferer);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new AuthenticationFailedException(e.getMessage(), e);
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: Add a proper error message here than getting from e

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 22]
```diff
@@ -0,0 +1,240 @@
...
+
+    /**
+     * Process the response of the Inwebo end-point
+     */
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+        if (pushResponse.contains(InweboConstants.PUSHRESPONSE)) {
+            log.info("Authentication Successful");
+            context.setSubject("Success");
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: This is incorrect. Subject should be an user attribute defined in IDP configuration.

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 23]
```diff
@@ -0,0 +1,240 @@
...
+            Map<String, Object> userClaims = getUserClaims();
+            if (userClaims != null && !userClaims.isEmpty()) {
+                context.setSubjectAttributes(getSubjectAttributes(userClaims));
+                context.setSubject("Authentication Success");
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: This is incorrect. Subject should be an user attribute defined in IDP configuration.

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 24]
```diff
@@ -0,0 +1,240 @@
...
+            throws AuthenticationFailedException {
+        try {
+            if (pushResponse == null) {
+                Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+                if (authenticatorProperties != null) {
+                    String relyingParty = request.getParameterValues("relyingParty")[0];
+                    String type = request.getParameterValues("type")[0];
+                    String referer = request.getHeader("referer").toString();
+                    String finalReferer = "";
+                    if ((referer.charAt(referer.lastIndexOf(":") + 5) + "").equals("/")) {
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: Why this logic is so hard coded with counting the index in exact numbers? It's not good to have it like this as even a slight change will break the flow.

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 25]
```diff
@@ -0,0 +1,240 @@
...
+        retryCount.setDescription("Number of attempts waiting for authentication(<10)");
+        configProperties.add(retryCount);
+
+        Property retryInterval = new Property();
+        retryInterval.setName(InweboConstants.RETRINTERVAL);
+        retryInterval.setDisplayName("Retry Interval");
+        retryInterval.setRequired(true);
+        retryInterval.setDescription("Period of time for waiting eg:1000");
+        configProperties.add(retryInterval);
+        return configProperties;
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: Adhere to a convention on the Displayname and description. (either make first letter in each word capital in all or not.) UI needs to be consistent.

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 26]
```diff
@@ -0,0 +1,240 @@
...
+            }
+        }
+
+        return claims;
+    }
+
+    protected Map<String, Object> getUserClaims() {
+        try {
+            String json = pushResponse;
+
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: remove unnecessary empty lines

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 27]
```diff
@@ -0,0 +1,240 @@
...
+            Map<String, Object> jsonObject = JSONUtils.parseJSON(json);
+
+            return jsonObject;
+        } catch (Exception e) {
+            log.error(e);
```
Comment (IsuraD) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: Need to throw exception here

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java [hunk 28]
```diff
@@ -0,0 +1,240 @@
...
+            }
+        } else
+            throw new AuthenticationFailedException("Authentication failed");
+        pushResponse = null;
+    }
+
+    protected Map<ClaimMapping, String> getSubjectAttributes(
+            Map<String, Object> claimMap) {
+
+        Map<ClaimMapping, String> claims = new HashMap<ClaimMapping, String>();
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java]: From Java - 1.7 onwards we can get rid of specifiying the generics at initiation.. 
new HashMap<ClaimMapping, String>()  --> new HashMap<>()
https://docs.oracle.com/javase/8/docs/technotes/guides/language/type-inference-generic-instance-creation.html

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushAuthenticate.java [hunk 29]
```diff
@@ -0,0 +1,131 @@
...
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushAuthenticate.java]: define specific classes to import than using *

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushAuthenticate.java [hunk 30]
```diff
@@ -0,0 +1,131 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+
+
+public class PushAuthenticate {
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushAuthenticate.java]: Not a suotable class name.. This is more like a method name.. Refer, http://www.oracle.com/technetwork/java/codeconventions-135099.html

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushAuthenticate.java [hunk 31]
```diff
@@ -0,0 +1,131 @@
...
+    private static final Log log = LogFactory.getLog(PushAuthenticate.class);
+
+    private static String urlString = "https://api.myinwebo.com/FS?";
+    private String serviceId;
+    private String p12file;
+    private String p12password;
+    SSLContext context = null;
+
+    public PushAuthenticate(String id, String p12file, String p12password) {
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushAuthenticate.java]: Not a suitable class name

File: inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushREST.java [hunk 32]
```diff
@@ -0,0 +1,106 @@
...
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONObject;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class PushREST {
```
Comment (Pushpalanka) [inline inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/PushREST.java]: Not a suitable class name

**Pull request closed without merging**